fs-storage/src/file_storage.rs

```diff
@@ -120,47 +182,32 @@ impl FileStorage {
         log::info!(
             "{} {} entries have been written",
             self.label,
-            value_by_id.len()
+            value_map.len()
         );
         Ok(())
     }
+}
 
-    /// Remove file at stored path
-    pub fn erase(&self) -> Result<()> {
-        fs::remove_file(&self.path).map_err(|err| {
-            ArklibError::Storage(self.label.clone(), err.to_string())
-        })
+impl<K, V> AsRef<BTreeMap<K, V>> for FileStorage<K, V> {
+    fn as_ref(&self) -> &BTreeMap<K, V> {
+        &self.data
     }
+}
 
-    /// Verify the version stored in the file header
-    fn verify_version(&self, header: &str) -> Result<()> {
-        if !header.starts_with(STORAGE_VERSION_PREFIX) {
-            return Err(ArklibError::Storage(
-                self.label.clone(),
-                "Unknown storage version prefix".to_owned(),
-            ));
-        }
-
-        let version = header[STORAGE_VERSION_PREFIX.len()..]
-            .parse::<i32>()
-            .map_err(|_err| {
-                ArklibError::Storage(
-                    self.label.clone(),
-                    "Failed to parse storage version".to_owned(),
-                )
-            })?;
+impl<K, V> Monoid<V> for FileStorage<K, V>
```

Comment:
The concept of "combining" two `FileStorage` structures needs a clear definition. Does it involve merging key-value pairs? How are conflicts handled? The current `combine` function only compares single values.

---
To get a sense of what I mean:

In a `Monoid` trait for integers, addition combines elements, with 0 as the neutral element. For example, combining `3` and `5` is `8`. This operation satisfies:
- Associativity: Grouping elements doesn't change the result (e.g., `(3 + 5) + 2` = `3 + (5 + 2)`).
- Neutral element: Adding 0 to any element leaves it unchanged (e.g., `3 + 0` = 3).

The `FileStorage<K, V>` struct holds a collection of key-value pairs. The `Monoid` trait doesn't naturally apply to this structure because:
What does combining two `FileStorage` instances even mean? Does it involve:
- Merging all key-value pairs?
- Overwriting existing keys with values from the second storage?
- How does it handle duplicate keys?

Reply:
We are implementing monoids for Tags, Properties, and Scores. While Tags and Properties are of type `HashSet`, they have already implements a union function. So, only the score left of type `INT` requires the particular function to be managed. That's why, instead of adding, we are comparing values(same as in Kotlin) - More info for scores [here](https://github.com/ARK-Builders/arklib-android/blob/main/lib/src/main/java/dev/arkbuilders/arklib/user/score/Score.kt#L9).

ref: https://github.com/ARK-Builders/ark-rust/issues/19#issuecomment-2018964840

Reply:
Thanks for the explanation, Pushkar.
I am still trying to wrap my head around the end goal.

I think adding a test would help clarify. 
Could you include a test where you create two `FileStorage` instances, each with its own tags, properties, and scores, and then use `FileStorage::combine_all()` to merge them?

Reply:
Yeah, that's for conflicts resolution. I'm not sure yet if `FileStorage` itself should be a `Monoid` (need to review the code for this).

What was done before in Kotlin is:
1. Take entries with unique keys from both maps, dump them into the resulting map.
2. For clashing entries, i.e. attached to same keys, we `combine` their values.

I think we can declare `FileStorage` as `Monoid`, if it's convenient. But it isn't necessary.